controllers/etcdcluster_controller.go

```diff
@@ -86,51 +115,131 @@ func (r *EtcdClusterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error)
 	err := cluster.ValidateCreate()
 	if err != nil {
 		log.Error(err, "invalid EtcdCluster")
-		return ctrl.Result{}, nil
+		return ctrl.Result{}, nil, nil
 	}
```

Comment:
I'd be inclined to keep the in-operator Defaulting and Validation check in the parent Reconcile function, so that we abandon both reconciliation and status updates if the custom resource is invalid.